build.sh

```diff
@@ -251,17 +263,27 @@ build_mscorlib()
 
     # The cross build generates a crossgen with the target architecture.
     if [ $__CrossBuild != 1 ]; then
-       if [ $__SkipCoreCLR == 0 -a -e $__BinDir/crossgen ]; then
-           echo "Generating native image for mscorlib."
-           $__BinDir/crossgen $__BinDir/mscorlib.dll
-           if [ $? -ne 0 ]; then
-               echo "Failed to generate native image for mscorlib."
-               exit 1
-           fi
+       # The architecture of host pc must be same architecture with target.
+       if [[ ( "$__HostArch" == "x64" ) && ( "$__BuildArch" == "x64" ) ]]; then
```

Comment:
We have made a simple condition at first as you described. We have updated the simple condition to handle both 32bit and 64bit architecture.